TuningApp.py

```diff
@@ -0,0 +1,592 @@
+            if self.temperature_tab1.text() in self.macbeth_used:
+                msg = QMessageBox()
+                msg.setText("Are you sure you want to overwrite image for this temperature?")
+                msg.setStandardButtons(QMessageBox.Ok | QMessageBox.Cancel)
+                ret = msg.exec()
+                if ret == QMessageBox.Ok:
+                    # macbeth procedure
+                    self.button_tab1.setText("Wait while detecting chart")
+                    self.button_tab1.setDisabled(True)
+                    temperature_value = self.temperature_tab1.text()
+                    lux_value = self.lux_tab1.text()
+                    self.macbeth_bool = True
+                    self.macbeth_used.append(temperature_value)
+                    # now we know that both of the values are intergers and macbeth plot is suppost to be in the picture
+                    adress = folder_directory + "/" + temperature_value + "K_" + lux_value + "L" + ".dng"
+                    cfg = picam2.create_still_configuration()
+                    picam2.switch_mode_and_capture_file(cfg, adress, signal_function=self.qpicamera2.signal_done, name="raw")
+                else:
+                    pass
+            else:
+                # macbeth procedure
+                self.button_tab1.setText("Wait while detecting chart")
+                self.button_tab1.setDisabled(True)
+                temperature_value = self.temperature_tab1.text()
+                lux_value = self.lux_tab1.text()
+                self.macbeth_bool = True
+                self.macbeth_used.append(temperature_value)
+                # now we know that both of the values are intergers and macbeth plot is suppost to be in the picture
+                adress = folder_directory + "/" + temperature_value + "K_" + lux_value + "L" + ".dng"
+                cfg = picam2.create_still_configuration()
+                picam2.switch_mode_and_capture_file(cfg, adress, signal_function=self.qpicamera2.signal_done, name="raw")
+
+        elif self.tabs.currentIndex() == 1:
+            # shading procedure
+            temperature_value = self.temperature_tab2.text()
+            self.alsc_used.append(temperature_value)
+            index = self.alsc_used.count(temperature_value)
+            adress = folder_directory + "/alsc_" + temperature_value + "K_" + str(index) + ".dng"
+            # update the lists
+            listWidgetItem_shading = QListWidgetItem("alsc_" + temperature_value + "K_" + str(index) + ".dng")
+            self.listWidget_shading.addItem(listWidgetItem_shading)
+            self.button_tab2.setEnabled(False)
+            cfg = picam2.create_still_configuration()
+            picam2.switch_mode_and_capture_file(cfg, adress, signal_function=self.qpicamera2.signal_done, name="raw")
+
+        else:
+            # cac procedure
+            self.cac_used += 1
+            adress = folder_directory + "/cac_chart" + str(self.cac_used) + ".dng"
```

Comment:
Maybe `filename` instead of `adress` (or `address`)? Don't know if `os.path.join` might make sense here? (Are we sure `folder_directory` doesn't end in a / ?)